src/ILCompiler/src/Program.cs

```diff
@@ -378,7 +378,14 @@ private int Run(string[] args)
                         inputModules.Add(module);
                     }
 
-                    compilationGroup = new MultiFileSharedCompilationModuleGroup(typeSystemContext, inputModules);
+                    if (_isReadyToRunCodeGen)
```

Comment:
Since multi-file is independent from ready-to-run, please pull the ready-to-run piece up out of the `if (_multifile)` block.